Automattic/wp-calypso
# Improve Stripe KYC flow for Paid Newsletters

**Note to self: Do not deploy until https://github.com/Automattic/jetpack/pull/29351 is released to WPCOM** 


Resolves https://github.com/Automattic/wp-calypso/issues/72282
Relates to  https://github.com/Automattic/jetpack/pull/29351 

## Proposed Changes

1.  When  query param, `?stripe_connect_success=earn-newsletter`, is available on the Earn page, display a Newsletter specific notice that prompt the user to create a newsletter payment plan after completing Stripe KYC

<img width="1093" alt="Screenshot 2023-03-06 at 9 54 03 PM" src="https://user-images.githubusercontent.com/7076981/223309166-c80e4b62-3979-487d-8ec9-4538bac8d6fb.png">

2. When "Add Payments" is clicked, open a new tab with the following URL: `earn/payments-plans/romarioraffearn.wordpress.com#add-newsletter-payment-plan,` which opens a modal to add a payment plan

<img width="928" alt="Screenshot 2023-03-06 at 9 53 42 PM" src="https://user-images.githubusercontent.com/7076981/223309482-7764e0ec-3e06-4d35-9241-15bd329c880d.png">


3. Update the "Add Payment Plan" modal header to be newsletter specific if the `#add-newsletter-payment-plan` hash is available. 

| Adding a Newsletter Plan  | Editing a Newsletter Plan |
| ------------- | ------------- |
| <img width="928" alt="Screenshot 2023-03-06 at 9 53 42 PM" src="https://user-images.githubusercontent.com/7076981/223309769-655cb164-34df-43e1-a225-d0bab36d3bab.png">  | <img width="931" alt="Screenshot 2023-03-06 at 9 53 50 PM" src="https://user-images.githubusercontent.com/7076981/223309772-243684d5-f92f-4d82-b14b-5bdf4d09e9d7.png">  |


## Testing Instructions



1. Go to `http://calypso.localhost:3000/earn/payments/romarioraffearn.wordpress.com?stripe_connect_success=earn-newsletter` and verify the notice with the following message is displayed: "Congrats! Your site is now connected to Stripe. You can now add payments to your newsletter."
2. Click on the "Add payments" CTA in the Notice
3. Verify that a new tab is opened and the modal is displayed to add a Newsletter Plan
4. Verify that the title of the modal is "Add a newsletter payment plan"
5. Create a newsletter payment plan
6. Edit the newsletter payment plan and verify the title of the modal is "Edit a newsletter payment plan"

### Demo

Appears that my microphone wasn't recording but the video should still be of value.

https://www.loom.com/share/cba348e10b4d4383aab2855ef1aa4822

## Pre-merge Checklist
